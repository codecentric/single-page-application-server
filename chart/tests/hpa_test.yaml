suite: test hpa
templates:
  - hpa.yaml
tests:
  - it: should not render hpa by default
    asserts:
      - hasDocuments:
          count: 0
  - it: should render hpa with default values if enabled
    values:
      - ./values/minimal_values.yaml
    set:
      pod.autoscaling.enabled: true
    asserts:
      - matchSnapshot: { }
  - it: should render hpa with custom metrics
    values:
      - ./values/minimal_values.yaml
    set:
      pod.autoscaling:
        enabled: true
        customMetrics:
          - type: Object
            object:
              metric:
                name: requests-per-second
              describedObject:
                apiVersion: networking.k8s.io/v1
                kind: Ingress
                name: some-ingress
              target:
                type: Value
                value: 10k
    asserts:
      - matchSnapshot: { }
  - it: should not render cpu metrics if targetCPUUtilizationPercentage is set to 0
    values:
      - ./values/minimal_values.yaml
    set:
      pod.autoscaling:
        enabled: true
        targetCPUUtilizationPercentage: 0
    asserts:
      - matchSnapshot: { }