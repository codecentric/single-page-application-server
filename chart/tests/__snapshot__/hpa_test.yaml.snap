should not render cpu metrics if targetCPUUtilizationPercentage is set to 0:
  1: |
    apiVersion: autoscaling/v2
    kind: HorizontalPodAutoscaler
    metadata:
      labels:
        app.kubernetes.io/component: frontend
        app.kubernetes.io/instance: default
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: RELEASE-NAME
        app.kubernetes.io/version: latest
        helm.sh/chart: single-page-application-server-0.0.0
      name: RELEASE-NAME
    spec:
      maxReplicas: 4
      metrics: null
      minReplicas: 1
      scaleTargetRef:
        apiVersion: apps/v1
        kind: Deployment
        name: RELEASE-NAME
should render hpa with custom metrics:
  1: |
    apiVersion: autoscaling/v2
    kind: HorizontalPodAutoscaler
    metadata:
      labels:
        app.kubernetes.io/component: frontend
        app.kubernetes.io/instance: default
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: RELEASE-NAME
        app.kubernetes.io/version: latest
        helm.sh/chart: single-page-application-server-0.0.0
      name: RELEASE-NAME
    spec:
      maxReplicas: 4
      metrics:
        - resource:
            name: cpu
            target:
              averageUtilization: 50
              type: Utilization
          type: Resource
        - object:
            describedObject:
              apiVersion: networking.k8s.io/v1
              kind: Ingress
              name: some-ingress
            metric:
              name: requests-per-second
            target:
              type: Value
              value: 10k
          type: Object
      minReplicas: 1
      scaleTargetRef:
        apiVersion: apps/v1
        kind: Deployment
        name: RELEASE-NAME
should render hpa with default values if enabled:
  1: |
    apiVersion: autoscaling/v2
    kind: HorizontalPodAutoscaler
    metadata:
      labels:
        app.kubernetes.io/component: frontend
        app.kubernetes.io/instance: default
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: RELEASE-NAME
        app.kubernetes.io/version: latest
        helm.sh/chart: single-page-application-server-0.0.0
      name: RELEASE-NAME
    spec:
      maxReplicas: 4
      metrics:
        - resource:
            name: cpu
            target:
              averageUtilization: 50
              type: Utilization
          type: Resource
      minReplicas: 1
      scaleTargetRef:
        apiVersion: apps/v1
        kind: Deployment
        name: RELEASE-NAME
